include/swift/SIL/SILDebugInfoExpression.h

```diff
@@ -74,6 +82,7 @@ struct SILDIExprElement {
     SILDIExprOperator Operator;
     Decl *Declaration;
     uint64_t ConstantInt;
+    Type TypePtr;
```

Comment:
Why is a TypePtr needed for tuples, but not for structs?
Couldn't we get it from the `Declaration`?
Is it because tuples are not Nominal types?

Reply:
Yes, for structs we use a VarDecl, but for tuples we don't have one, they have no declaration

Reply:
I understand that we don't have a TypeDecl, but we should still have a VarDecl. That said, I guess this is really supposed to be a type decl.
Could we use a `Type` in the struct case as well?

Reply:
The VarDecl for the struct case refers to the struct member, whose parent is the struct type. The tuple has no VarDecl as the members are never declared anywhere

Reply:
Potentially we can store both in an `llvm::PointerUnion<Decl, TypeBase>`?

Reply:
Nevermind, this is already a union.